File: frontend/src/services/knowledgeBase.ts
import { api } from './api'

export interface KnowledgeBaseNoteItem {
  path: string
  title: string
  size: number
  updated_at: string
}

export interface KnowledgeBaseNoteListResponse {
  items: KnowledgeBaseNoteItem[]
  total: number
  root_exists: boolean
}

export interface KnowledgeBaseNoteContent {
  path: string
  title: string
  content_markdown: string
}

export interface KnowledgeBasePriorResponse {
  query: string
  prior_summary: string
  has_prior: boolean
}

function extractErrorMessage(error: any, fallback: string): string {
  const detail = error?.response?.data?.detail
  if (typeof detail === 'string') return detail
  if (detail && typeof detail === 'object' && typeof detail.detail === 'string') return detail.detail
  return fallback
}

class KnowledgeBaseService {
  private readonly basePath = '/knowledge-base'

  async listNotes(): Promise<KnowledgeBaseNoteListResponse> {
    try {
      return await api.get<KnowledgeBaseNoteListResponse>(`${this.basePath}/notes`)
    } catch (error: any) {
      throw new Error(extractErrorMessage(error, '加载知识库失败'))
    }
  }

  async getNote(path: string): Promise<KnowledgeBaseNoteContent> {
    try {
      return await api.get<KnowledgeBaseNoteContent>(`${this.basePath}/notes/content`, {
        params: { path },
      })
    } catch (error: any) {
      throw new Error(extractErrorMessage(error, '加载笔记失败'))
    }
  }

  async getPrior(query: string): Promise<KnowledgeBasePriorResponse> {
    try {
      return await api.get<KnowledgeBasePriorResponse>(`${this.basePath}/prior`, {
        params: { q: query },
      })
    } catch (error: any) {
      throw new Error(extractErrorMessage(error, '查询先前知识失败'))
    }
  }
}

export const knowledgeBaseService = new KnowledgeBaseService()
export default knowledgeBaseService

File: backend/app/schemas/knowledge_base.py
"""
学生个人知识库 Schema
"""

from __future__ import annotations

from datetime import datetime
from typing import List

from pydantic import BaseModel, Field


class KnowledgeBaseNoteItem(BaseModel):
    path: str
    title: str
    size: int
    updated_at: datetime


class KnowledgeBaseNoteListResponse(BaseModel):
    items: List[KnowledgeBaseNoteItem] = Field(default_factory=list)
    total: int = 0
    root_exists: bool = False


class KnowledgeBaseNoteContent(BaseModel):
    path: str
    title: str
    content_markdown: str


class KnowledgeBasePriorResponse(BaseModel):
    query: str
    prior_summary: str
    has_prior: bool = False
